libgit/worktree_storer.go

```diff
@@ -0,0 +1,72 @@
+// Copyright 2017 Keybase Inc. All rights reserved.
+// Use of this source code is governed by a BSD
+// license that can be found in the LICENSE file.
+
+package libgit
+
+import (
+	"github.com/pkg/errors"
+	"gopkg.in/src-d/go-git.v4/plumbing"
+	"gopkg.in/src-d/go-git.v4/plumbing/format/index"
+	"gopkg.in/src-d/go-git.v4/plumbing/storer"
+	"gopkg.in/src-d/go-git.v4/storage"
+)
+
+// worktreeStorer fake out the storage layer in such a way that it
+// stores its index, and any reference modifications, in a .git/
+// directory in the worktree, but everything else goes through the
+// main bare repo's .git directory.  This is useful for making a quick
```

Comment:
Can you give an example? I'm having trouble visualizing what this is describing.